packages/ui/react-ui-deck/src/components/Deck/Deck.tsx

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import React, { type ComponentPropsWithRef, forwardRef, useCallback, useEffect, useState } from 'react';
+
+import { Main, type MainProps, type ThemedClassName, useMediaQuery, useTranslation } from '@dxos/react-ui';
+import { mx } from '@dxos/react-ui-theme';
+
+import { translationKey } from '../../translations';
+
+type DeckRootProps = MainProps;
+
+const deckLayout =
+  'fixed inset-0 z-0 overflow-x-auto overflow-y-hidden snap-inline snap-proximity grid grid-rows-[var(--rail-size)_[toolbar-start]_var(--rail-action)_[content-start]_1fr_[content-end]] grid-cols-[repeat(99,min-content)]';
+
+const resizeButtonStyles =
+  'hidden sm:grid is-6 row-span-3 p-0 cursor-col-resize touch-none justify-items-center items-stretch plb-2 before:rounded-full before:is-1 before:surface-input hover:before:surface-inputHover before:transition-colors hover:before:surface-accentFocusIndicator focus-visible:before:surface-accentFocusIndicator data-[resizing=true]:before:surface-accentFocusIndicator focus:outline-none';
+
+const DeckRoot = forwardRef<HTMLDivElement, DeckRootProps>(({ classNames, children, ...props }, forwardedRef) => {
+  return (
+    <Main.Content {...props} classNames={[deckLayout, classNames]} ref={forwardedRef}>
+      {children}
+    </Main.Content>
+  );
+});
+
+type DeckPlankUnit = 'rem' | 'px';
+
+type DeckPlankProps = ThemedClassName<ComponentPropsWithRef<'article'>> & { unit?: DeckPlankUnit };
+
+type DeckPlankResizing = Pick<MouseEvent, 'pageX'> & { size: number } & { [Unit in DeckPlankUnit]: number };
+
+const DeckPlank = forwardRef<HTMLDivElement, DeckPlankProps>(
+  ({ unit = 'rem', classNames, style, children, ...props }, forwardedRef) => {
+    const { t } = useTranslation(translationKey);
+    const [isSm] = useMediaQuery('sm', { ssr: false });
+
+    const [size, setSize] = useState<number>(40);
```

Comment:
do we have consts for density relevant variables.

Reply:
My understanding was that density, where @dxos/react-ui is concerned, is about vertical cadence, not width. Did you anticipate 'coarse' or 'fine' meaning something where Deck is concerned?

fwiw this is basically a placeholder, I imagine this will become a controllable state with lots more nuance provided by the plugins, e.g. markdown documents and chats would have fairly different defaults.